internal/bminventory/inventory.go

```diff
@@ -1609,9 +1609,23 @@ func filterReply(expected interface{}, input string) (string, error) {
 func (b *bareMetalInventory) DisableHost(ctx context.Context, params installer.DisableHostParams) middleware.Responder {
 	log := logutil.FromContext(ctx, b.log)
 	var host models.Host
+	var cluster common.Cluster
 	log.Info("disabling host: ", params.HostID)
 
-	if err := b.db.First(&host, identity.AddUserFilter(ctx, "id = ? and cluster_id = ?"), params.HostID, params.ClusterID).Error; err != nil {
+	txSuccess := false
```

Comment:
i can understand the transaction for the disable part. but why do you need a transaction for the refresh?